catalog/ui/catalog-ui-search/src/main/webapp/js/controllers/cesium.layerCollection.controller.js

```diff
@@ -96,10 +96,14 @@ define(['underscore',
         setAlpha: function (model) {
             var layer = this.layerForCid[model.id];
             layer.alpha = model.get('alpha');
+            layer.show = this.showLayer(model);
         },
         setShow: function (model) {
             var layer = this.layerForCid[model.id];
-            layer.show = model.get('show');
+            layer.show = this.showLayer(model);
+        },
+        showLayer: function(model) {
```

Comment:
Let's move this method to be available on the layer model in `User.js`.   See https://github.com/codice/ddf/blob/master/catalog/ui/catalog-ui-search/src/main/webapp/js/model/User.js#L90 for where to move it.  

Might also be worth renaming it to `shouldShowLayer`.